### ISSUE TITLE
InvocationResponseTypeDef: incompatible type "bytes"; expected "SupportsRead[Union[str, bytes]]"[BUG]

### ISSUE THREAD

**Issue by tobwiens** [vemel/mypy_boto3_builder#52]
**Describe the bug**
It seems that the InvocationResponseTypeDef ['Payload'] has the type bytes which triggers the type checker.

`incompatible type "bytes"; expected "SupportsRead[Union[str, bytes]]"
                return json.load(invoke_response['Payload'])`

**To Reproduce**
1. Install 'boto3-stubs[lambda]'
2.  Run mypy on the following code sample

```python
import boto3
lambda_payload = {"test" : 123}
invoke_response: InvocationResponseTypeDef = boto3.client('lambda')invoke(FunctionName="some-lambda",
                                                    Payload=lambda_payload.encode('utf-8'))`
response_payload = json.load(invoke_response['Payload])


```
3. Actual mypy output

```
incompatible type "bytes"; expected "SupportsRead[Union[str, bytes]]"
                return json.load(invoke_response['Payload'])
```

**Expected mypy-output**

```
...
```

**Additional context**
`pip3.8 install boto3-stubs[essential]==1.16.2.0

The type definition defines the Payload as bytes, the [boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/1.15.16/reference/services/lambda.html#Lambda.Client.invoke) describes it as StreamingBody()  
```
InvocationResponseTypeDef = TypedDict(
    "InvocationResponseTypeDef",
    {
        "StatusCode": int,
        "FunctionError": str,
        "LogResult": str,
        "Payload": bytes,
        "ExecutedVersion": str,
    },
    total=False,
)
```

From what I understood reading other PRs it seems to me that the Payload should be of type `IO[bytes]`
```
InvocationResponseTypeDef = TypedDict(
    "InvocationResponseTypeDef",
    {
        "StatusCode": int,
        "FunctionError": str,
        "LogResult": str,
        "Payload": IO[bytes],
        "ExecutedVersion": str,
    },
    total=False,
)
```

**Comment by vemel** [comment]
Thank you for the report. Yes, looks like boto3 overrides the original schema. I will add it to overrides.

**Comment by vemel** [comment]
This is a misconfiguration in `botocore` shapes. `InvocationResponse.Payload` is a `blob`, but it is not defined as `streaming: True`.  Looks like the only way is to hardcode, because parser logic looks correct.

**Comment by vemel** [comment]
@tobwiens fixed in `3.2.2` (`boto3-stubs>=1.16.5.1`). Please let me know if it works as it should for you.

**Comment by tobwiens** [comment]
@vemel Works great, thank you for the fast fix.
